cmd/agent/app/integrations.go

```diff
@@ -391,6 +391,9 @@ func downloadWheel(integration, version string) (string, error) {
 		args = append(args, fmt.Sprintf("-%s", strings.Repeat("v", verbose)))
 	}
 	downloaderCmd := exec.Command(pyPath, args...)
+	// Change the working directory to one the Datadog Agent can read, so that we
+	// can switch to temporary working directories, and back, for in-toto.
+	downloaderCmd.Dir, _ = executable.Folder()
```

Comment:
I'm not sure this (`/opt/datadog-agent/bin/agent`) would be the ideal location for temp work. I would probably prefer using `/tmp` or `/opt/datadog-agent/run` (and of course something suitable on windows too).

Maybe I'm missing some context?

Reply:
So the underlying downloader, a python script will be doing this: https://github.com/DataDog/integrations-core/blob/master/datadog_checks_downloader/datadog_checks/downloader/download.py#L166-L197

As we can see, there we're calling `tempfile.mkdtemp()` (with no function arguments).  And therefore no `dir` parameter specified so, as per the python docs:
```
If dir is specified, the file will be created in that directory; otherwise, a default directory is used. The default directory is chosen from a platform-dependent list, but the user of the application can control the directory location by setting the TMPDIR, TEMP or TMP environment variables. There is thus no guarantee that the generated filename will have any nice properties, such as not requiring quoting when passed to external commands via os.popen().
```

The temporary directory will be created in a pre-defined location (`/tmp` most likely on linux) which should be writable by the `dd-agent` user. It doesn't matter if we `chdir` somewhere else, the temporary directory would still be created in that predefined location. If we want to have finer control over where we're going to create this temporary work directory, we could use one of the `TMPDIR`, `TEMP` or `TMP` environment variables to override to a better location. 

So, all that being said, I'm not sure changing directories here adds value.